docs/source/app_developers_guide/zmq_event_subscription.rst

```diff
@@ -0,0 +1,309 @@
+        subscriptions=[subscription]).SerializeToString()
+
+    # Construct the message wrapper
+    correlation_id = "123" # This must be unique for all in-flight requests
+    msg = Message(
+        correlation_id=correlation_id,
+        message_type=CLIENT_EVENTS_SUBSCRIBE_REQUEST,
+        content=request)
+
+    # Send the request
+    socket.send_multipart([msg.SerializeToString()])
+
+.. note::
+
+   For information about ``ClientEventsSubscribeRequest``, see
+   :doc:`about_event_subscriptions`.
+
+
+Step 3: Receiving the Response
+==============================
+
+After submitting the subscription request, wait for a response from the
+validator. The validator will return a response indicating whether the
+subscription was successful.
+
+The following example receives the response and verifies the status.
+
+.. code-block:: python
+
+    # Receive the response
+    resp = socket.recv_multipart()[-1]
+
+    # Parse the message wrapper
+    msg = Message()
+    msg.ParseFromString(resp)
+
+    # Validate the response type
+    if msg.message_type != CLIENT_EVENTS_SUBSCRIBE_RESPONSE:
+        print("Unexpected message type")
+
+    # Parse the response
+    response = ClientEventsSubscribeResponse()
+    response.ParseFromString(msg.content)
+
+    # Validate the response status
+    if response.status != ClientEventsSubscribeResponse.OK:
+      print("Subscription failed: {}".format(response.response_message))
+
+
+Step 4: Listening for Events
+============================
+
+After the event subscription request has been sent and accepted, events will
+arrive on the ZMQ socket. The application must start listening for these events.
+
+.. note::
+
+   In order to limit network traffic, individual events are wrapped in an
+   event list message before being sent.
+
+The following example listens for events and prints them indefinitely.
+
+.. code-block:: python
+
+    while True:
+      resp = socket.recv_multipart()[-1]
+
+      # Parse the message wrapper
+      msg = Message()
+      msg.ParseFromString(resp)
+
+      # Validate the response type
+      if msg.message_type != CLIENT_EVENTS:
+          print("Unexpected message type")
+
+      # Parse the response
+      events = EventList()
+      events.ParseFromString(msg.content)
+
+      for event in events:
+        print(event)
+
+
+.. _correlate-events-to-blocks-label:
+
+Correlating Events to Blocks
+============================
+
+An event originates from a specific block. That is, an event is sent to the
+subscriber only when the block is committed and state is updated. As a result,
+events can be treated as output from processing and committing blocks.
+
+An application can subscribe to both ``sawtooth/block-commit`` and
+``sawtooth/state-delta`` events to match state changes with the block
+in which the changes occurred.
+
+All lists of ``block-commit`` events received from the validator will contain a
+``block-commit`` event for the block that the events came from.
```

Comment:
Is there a typo here? I don't know how to interpret this sentence.

Reply:
There's no typo, it's describing the contents of a block commit event. It might need a "contain only a single ``block-commit`` event" as part of its sentence.

Reply:
I propose changing this sentence to:

 "All lists of ``block-commit`` events received from the validator will contain only a single ``block-commit`` event for the block that the events came from."

Does that sound correct?

Reply:
Unless this is supposed to mean "no matter how many events were generated when a block is committed, each block-commit event received from the validator is a single event"?  Because I don't know how to interpret this sentence, either.  I'm confused by "All lists of block-commit events" - if a list always contains only a single event, it's not really list.